cmd/db-migration/db-migration.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/drand/drand-tools/internal/migration"
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/fs"
+	"github.com/drand/drand/log"
+)
+
+var sourcePath = flag.String("source", "", "The source database to be migrated to the new format.")
+var beaconName = flag.String("beacon", "", "The name of the beacon to be migrated.")
+var migrationTarget = flag.String("target", "", "The type of database to migrate to. Supported values: boltdb, postgres.\n"+
+	"If boltdb is used, then the migration will be done in-place.\n"+
+	"If postgres is used, then you need to specify the -pg-dsn flag value",
+)
+
+//nolint:lll // This is a flag
+var pgDSN = flag.String("pg-dsn", `postgres://drand:drand@localhost:5432/drand?sslmode=disable&connect_timeout=5`, "The connection details for Postgres.")
+
+//nolint:lll // This is a flag
+var bufferSize = flag.Int("buffer-size", migration.DefaultBufferSize, "Number of beacons that can be migrated at once. Use 0 to pre-allocate all the beacons.")
+
+func main() {
+	flag.Parse()
+	logger := log.NewLogger(nil, log.LogDebug)
+
+	err := checkValues(*sourcePath, *beaconName, *pgDSN, chain.StorageType(*migrationTarget))
+	if err != nil {
+		logger.Panicw("while validating input values", "err", err)
+	}
+
+	err = migrate(logger, *sourcePath, *beaconName, *pgDSN, chain.StorageType(*migrationTarget), *bufferSize)
+	if err != nil &&
+		!errors.Is(err, migration.ErrMigrationNotNeeded) {
+		logger.Panicw("while migrating the databaase", "err", err)
```

Comment:
Could we move all this code to a run function and return an error back to main. Then we can fmt this and os.Exit(1)